src/main/java/com/searchgears/queryanalysis/matching/DictionaryRewriterHolder.java

```diff
@@ -75,6 +84,16 @@ private void createDictionaryRewriter() {
         this.rewriter = new DictionaryRewriter(synFilepath.getFileName().toString(), loader);
     }
 
+    private void deleteTempSynFile() {
+        if (deleteSynFile) {
```

Comment:
Personally, I would prefer the if to be outside or the name of the method to reflect that it might also _not_ delete the syn file :-)

Reply:
Moved the `if` statement to the top level method.